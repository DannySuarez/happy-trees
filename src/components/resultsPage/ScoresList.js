import React, { useState, useEffect } from 'react';
import Score from './Score';
import PropTypes from 'prop-types';
import getScores from '../../services/getScoresApi';
import { Link } from 'react-router-dom';


const ScoresList = () => {

  const [scores, setScores] = useState([]);

  useEffect(() => {
    getScores()
      .then(setScores).catch(alert);
  });

  const scoresList = scores.map(score => {
    return <Score nickName={score.nickName} score={score.score} key={score.id} />;
  });

  return (
      <>
      <h1>Game Results</h1>
      <div>
        <h3>Scores</h3>
        {scoresList}
      </div>
      <div>
        {/* fix buttons to link to home page and landingPage */}
        <Link to="/">
          <button>Home</button>
        </Link>
        <button>Play Again</button>
      </div>
      </>
  );
};

ScoresList.propTypes = {
  scores: PropTypes.array
};

export default ScoresList;
