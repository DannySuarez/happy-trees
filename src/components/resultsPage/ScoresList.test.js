import React from 'react';
import { shallow } from 'enzyme';
import ScoresList from './ScoresList';

describe('ScoresList Input', () => {
  it('renders ScoresList component', () => {
    const wrapper = shallow(<ScoresList />);
    expect(wrapper).toMatchSnapshot();
  });
});
