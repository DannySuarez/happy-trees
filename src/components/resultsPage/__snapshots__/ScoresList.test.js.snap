// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoresList Input renders ScoresList component 1`] = `
<Fragment>
  <h1>
    Game Results
  </h1>
  <div>
    <h3>
      Scores
    </h3>
  </div>
  <div>
    <Link
      to="/"
    >
      <button>
        Home
      </button>
    </Link>
    <button>
      Play Again
    </button>
  </div>
</Fragment>
`;
